dpctl/tensor/_copy_utils.py

```diff
@@ -739,6 +739,9 @@ def astype(
             order=copy_order,
             buffer_ctor_kwargs={"queue": usm_ary.sycl_queue},
         )
+    # see #2121
+    if ary_dtype == dpt.bool:
+        usm_ary = dpt.not_equal(usm_ary, 0, order=order)
```

Comment:
The issue is resolved. Thank you @ndgrigorian!
The only question is about `order` keyword, why do we pass it as `order=order` and would it be more efficient to use `order=copy_order` instead to be aligned with order of `R` array?